### PR TITLE
Various MMF bug fixes

### PR DESCRIPTION
**Pull request by whannah1**
Fix a few minor issues with the MMF:
- revive standalone test framework
- aerosol optics syntax bug
- YAKL updates

[BFB]

### CODE REVIEW

File: components/cam/src/physics/crm/RNG_MT.F90 [hunk 1]
```diff
@@ -50,9 +50,8 @@
 !-----------------------------------------------------------------------
 !-----------------------------------------------------------------------
 module RNG_MT
-   use shr_kind_mod,  only: r8 => shr_kind_r8
-
    implicit none
+   integer, parameter :: r8 = selected_real_kind(13)
```
Comment (brhillman) [inline components/cam/src/physics/crm/RNG_MT.F90:54]: Hard-coding this is to remove dependency on shr_kind_mod?

File: components/cam/src/physics/crm/openacc_utils.F90 [hunk 2]
```diff
@@ -8,7 +8,7 @@ module openacc_utils
   use cudafor
 #endif
   implicit none
-  integer :: ierr
+  integer, private :: ierr
```
Comment (brhillman) [inline components/cam/src/physics/crm/openacc_utils.F90:11]: Usually prefer to see "private" by default for everything in the module, and then explicitly make certain variables/routines public. Is there a reason to instead assume everything is public here?

File: components/cam/src/physics/crm/sam/accelerate_crm.F90 [hunk 3]
```diff
@@ -20,11 +20,10 @@
 ! -----------------------------------------------------------------------------
 module accelerate_crm_mod
     use grid, only: nx, ny
-    use shr_kind_mod, only: r8=>shr_kind_r8
     use params, only: asyncid, rc=>crm_rknd
-
     implicit none
     public
+    integer, parameter :: r8 = selected_real_kind(13)
```
Comment (brhillman) [inline components/cam/src/physics/crm/sam/accelerate_crm.F90:26]: I still think these type defs should be in a centralized place, maybe in `params` if not adopted from `shr_kind_mod`?

File: components/cam/src/physics/crm/openacc_utils.F90 (hunk 2 above)
Comment (rljacob) [reply components/cam/src/physics/crm/openacc_utils.F90:11]: Yes all modules should have "private" as default.

File: components/cam/src/physics/crm/sam/accelerate_crm.F90 (hunk 3 above)
Comment (rljacob) [reply components/cam/src/physics/crm/sam/accelerate_crm.F90:26]: Yes use the kind modifiers provided by shr_kind_mod.

Comment (whannah1) [reply components/cam/src/physics/crm/sam/accelerate_crm.F90]: Ben, this should be resolved in the latest changes.

File: components/cam/src/physics/crm/openacc_utils.F90 (hunk 2 above)
Comment (whannah1) [reply components/cam/src/physics/crm/openacc_utils.F90]: addressed in recent commit

File: components/cam/src/physics/crm/sam/MICRO_M2005/drop_activation.F90 [hunk 4]
```diff
@@ -13,9 +13,8 @@ module drop_activation
 ! July, 2009:  adopted from the module of ndrop used in CAM. 
 !
 !---------------------------------------------------------------------------------------------------- 
-   use shr_kind_mod, only: r8 => shr_kind_r8
    use modal_aero_data, only: ntot_amode
-   use params, only: crm_rknd
+   use params_kind
```
Comment (brhillman) [inline components/cam/src/physics/crm/sam/MICRO_M2005/drop_activation.F90:17]: Can we state explicitly which types from params_kind we are using here?

File: components/cam/src/physics/crm/sam/accelerate_crm.F90 [hunk 5]
```diff
@@ -20,19 +20,18 @@
 ! -----------------------------------------------------------------------------
 module accelerate_crm_mod
     use grid, only: nx, ny
-    use shr_kind_mod, only: r8=>shr_kind_r8
-    use params, only: asyncid, rc=>crm_rknd
-
+    use params, only: asyncid
+    use params_kind
```
Comment (brhillman) [inline components/cam/src/physics/crm/sam/accelerate_crm.F90:24]: Can we say `use params_kind, only: crm_rknd` here for clarity?

File: components/cam/src/physics/crm/sam/crm_module.F90 [hunk 6]
```diff
@@ -24,6 +24,7 @@ module crm_module
   use damping_mod
   use ice_fall_mod
   use coriolis_mod
+  use params_kind
```
Comment (brhillman) [inline components/cam/src/physics/crm/sam/crm_module.F90:27]: As above, can we say `use params_kind, only: crm_rknd` here?

File: components/cam/src/physics/crm/RNG_MT.F90 (hunk 1 above)
Comment (brhillman) [inline components/cam/src/physics/crm/RNG_MT.F90:54]: I think here we should use `use params_kind, only: r8`

File: components/cam/src/physics/crm/crm_rad_module.F90 [hunk 7]
```diff
@@ -8,9 +8,9 @@ module crm_rad_module
    ! update the radiation in the future, should we choose to put the radiation
    ! calculations on the CRM.
    use params_kind,       only: crm_rknd
-   use shr_kind_mod, only: r8 => shr_kind_r8
 
    implicit none
+   integer, parameter :: r8 = selected_real_kind(13)
```
Comment (brhillman) [inline components/cam/src/physics/crm/crm_rad_module.F90:13]: I think now that we have the single definition of `r8`, we should use `use params_kind, only: crm_rknd, r8` in the above use statement and eliminate this definition.

**Review by brhillman — APPROVED** [review]

**Comment by brhillman** [comment]
Merged to next.

**Comment by brhillman** [comment]
Merged to master.

**Pull request merged**